Tests/MongoSwiftSyncTests/SpecTestRunner/TestOperationResult.swift

```diff
@@ -71,12 +84,33 @@ enum TestOperationResult: Decodable, Equatable {
             return lhsArray == rhsArray
         case let (.document(lhsDoc), .document(rhsDoc)):
             return lhsDoc.sortedEquals(rhsDoc)
+        case let (.error(lhsErr), .error(rhsErr)):
+            return lhsErr == rhsErr
+        default:
+            return false
+        }
+    }
+
+    internal func contentMatches(expected: TestOperationResult) -> Bool {
+        switch (self, expected) {
+        case let (.bulkWrite(bw), .bulkWrite(expectedBw)):
+            return bw.matches(expected: expectedBw)
+        case let (.int(int), .int(expectedInt)):
+            return int.matches(expected: expectedInt)
+        case let (.array(array), .array(expectedArray)):
+            return array.matches(expected: expectedArray)
+        case let (.document(doc), .document(expectedDoc)):
+            return doc.matches(expected: expectedDoc)
+        case let (.error(error), .error(expectedError)):
+            return error.matches(expected: expectedError)
```

Comment:
we actually don't want to use the `Matchable` conformance for handling errors, we want to use the other custom logic in `checkErrorResult`. So we should just return false in the `(.error, _)` case.